### PR TITLE
[Chore] Upgrade to Symfony 6.2

### PR DESCRIPTION
**Pull request by IonBazan**
Upgrading to Symfony 6.2 with following steps:

- [x] Bumping Symfony dependencies to `^6.2`
- [x] Bumping Rector to v0.15 and PHPStan to v1.9.2 (to utilize Rector presets)
- [x] Running Rector with `SYMFONY_62` preset
- [x] Removing `sensio/framework-extra-bundle` (all used attributes are now part of Symfony core)
- [x] Removing deprecated `security.enable_authenticator_manager` config
- [x] Refactoring `ArgumentValueResolver`s into `ValueResolver`s
- [x] [Minor] Running CS Fixer
- [x] [Minor] Bump `phpstan-doctrine` and `php-symfony`

See https://github.com/symfony/symfony/blob/6.2/UPGRADE-6.2.md and https://github.com/rectorphp/rector-symfony/blob/main/config/sets/symfony/symfony62.php

### CODE REVIEW

File: phpstan.neon [hunk 1]
```diff
@@ -38,6 +38,7 @@ parameters:
         - '~Property App\\Entity\\PhpStat::\$type type mapping mismatch: database can contain int but property expects~'
         - '~Property App\\Entity\\PhpStat::\$depth type mapping mismatch: database can contain int but property expects~'
         - '~Property App\\Entity\\Job::\$status type mapping mismatch: database can contain string but property expects~'
+        - '~Property App\\Entity\\\w+::\$\w+ type mapping mismatch: property can contain Doctrine\\Common\\Collections\\Collection~'
```
[the remark this answers is not in the document]
Comment (IonBazan) [reply phpstan.neon:41]: It's reporting a bunch of errors on relationships:
>Property App\Entity\SecurityAdvisory::$sources type mapping mismatch: property can contain Doctrine\Common\Collections\Collection<int, App\Entity\SecurityAdvisorySource>&Doctrine\Common\Collections\Selectable<int, App\Entity\SecurityAdvisorySource> but database expects
         Doctrine\Common\Collections\Collection&iterable<App\Entity\SecurityAdvisorySource>.

[the remark this answers is not in the document]
Comment (stof) [reply phpstan.neon:41]: this looks like a bug in phpstan-doctrine, that should be reported upstream

[the remark this answers is not in the document]
Comment (IonBazan) [reply phpstan.neon]: Bumped `phpstan-doctrine` and it's fixed now 👍🏻

File: src/ArgumentResolver/PackageResolver.php [hunk 2]
```diff
@@ -16,27 +16,25 @@
...
-        return Package::class === $argument->getType();
     }
 
     /**
      * @return iterable<Package>
      */
     public function resolve(Request $request, ArgumentMetadata $argument): iterable
     {
+        if (Package::class !== $argument->getType()) {
+            return [];
```
Comment (stof) [inline src/ArgumentResolver/PackageResolver.php:35]: As this method is currently a generator, `return;` would be the way to make it return an empty iterable.

But my suggestion is to replace `yield $package;` with `return [$package];` as done in the core resolvers.

File: src/ArgumentResolver/UserResolver.php [hunk 3]
```diff
@@ -16,28 +16,26 @@
...
-        return User::class === $argument->getType() && \count($argument->getAttributes(CurrentUser::class, ArgumentMetadata::IS_INSTANCEOF)) === 0;
     }
 
     /**
      * @return iterable<User>
      */
     public function resolve(Request $request, ArgumentMetadata $argument): iterable
     {
+        if (User::class !== $argument->getType() || \count($argument->getAttributes(CurrentUser::class, ArgumentMetadata::IS_INSTANCEOF)) > 0) {
+            return [];
```
Comment (stof) [inline src/ArgumentResolver/UserResolver.php:36]: same here

**Comment by Seldaek** [comment]
Thanks

**Comment by Seldaek** [comment]
Had to revert the deploy of this upgrade as for some reason it is hitting the DB much harder than it should. I'm suspecting some caching somewhere isn't working like it should, but I haven't pinpointed the issue yet. Just FYI in case anyone has a clue.

**Comment by Seldaek** [comment]
I have narrowed it down to one of these:

  - Downgrading symfony/doctrine-bridge (v6.2.3 => v6.1.9)
  - Downgrading symfony/http-kernel (v6.2.4 => v6.1.10)
  - Downgrading symfony/mailer (v6.2.2 => v6.2.0)
  - Downgrading symfony/security-bundle (v6.2.3 => v6.1.9)
  - Downgrading symfony/security-http (v6.2.2 => v6.1.9)
  - Downgrading symfony/twig-bridge (v6.2.3 => v6.1.9)
  - Downgrading symfony/twig-bundle (v6.2.3 => v6.1.9)
  
Downgrading that fixes it. I doubt it's twig though just have to downgrade all these as a batch due to interdependencies.

Probably in http-kernel

**Comment by Seldaek** [comment]
Yeah that seems to be fine. So it's something in https://github.com/symfony/doctrine-bridge/compare/v6.1.9...v6.2.3 - either lazy loading ghosts, or the entity value resolver?

**Comment by Seldaek** [comment]
I tried disabling lazy ghosts AND making sure the entity resolver runs after our own PackageResolver here, but that didn't do the trick. I really have to run 6.1.9 for it to work well (see https://github.com/composer/packagist/commit/aa0b2bf8d23e0d38117b4b1b105d07d6af018562 for what's now in prod)

**Comment by IonBazan** [comment]
Thank you so much for merging this @Seldaek and investigating the issue. I'm glad you managed to isolate the problem and revert only singe package upgrade 👏🏻 
Interesting findings regarding the `EntityValueResolver` - maybe it's good to report it to Symfony team to keep track of the issue.

**Comment by stof** [comment]
No need to hack together. I can be disabled in config:

```
doctrine:
    orm:
        controller_resolver:
            enabled: false
```

**Comment by Seldaek** [comment]
@stof it was late and I didn't find that, but thanks that's more sustainable for a proper deploy :D 

Done in https://github.com/composer/packagist/commit/7debd9bde87bf3def3e316327e64a8fcf34388a8

**Comment by stof** [comment]
@Seldaek Note that if you have some cases where you still want to use the entity value resolver, my suggestion is to at least disable the `auto_mapping` feature. This would keep only the loading by primary key from an attribute named `id` or the same than the argument. Custom mappings are still possible by adding a `MapEntity` attribute on the argument to define an explicit mapping.
Your issue reported in https://github.com/symfony/symfony/pull/49034 would probably not happen without auto_mapping, as it would not use the `email` attribute to load a User object.

**Comment by Seldaek** [comment]
Sure there's an option but I don't think the UserValueResolver should be resolving random users out of the box that's a massive potential security issue. 

Anyway I found the root cause here and fixed it in https://github.com/composer/packagist/commit/35d9e2a00d513c542060836875d3a31cca260c95

In my urge to go bleeding edge I replaced the Doctrine ParamConverter a while back with PackageResolver and UserResolver, but now Symfony 6.2 restored automatic entity loading and this caused a little unforeseen issue.

```
#[Route(path: '/packages/{name}/stats/all.json', name: 'package_stats', requirements: ['name' => '[A-Za-z0-9_.-]+/[A-Za-z0-9_.-]+?'])]
public function overallStatsAction(Request $req, #[VarName('name')] Package $package, ?Version $version = null, ?string $majorVersion = null): JsonResponse
```

In this action is a nullable `Version $version`, but as the `name` property exists on `Version` the `EntityValueResolver` tried to load a `Version` with `findOneBy(['name' => 'pkg/name'])` whenever you loaded stats (which happens a lot due to crawlers and others..). There is no index on the name column on the package_version table because we pretty much never need to query those by name. So it ended up scanning the whole table which our largest table resulting in massive IO on the database server.

Mystery resolved, code is refactored now to avoid these useless args on the action which was to be fair sloppy.

**Comment by Seldaek** [comment]
Right, I think it's ok as is.. it's really just a problem because the codebase here did not expect it, and wasn't built with entity loading in mind. For most projects migrating from DoctrineParamConverter to the new EntityValueResolver I am assuming they'll be happy with it.

**Comment by stof** [comment]
Most project probably load their entities by primary key without even realizing that DoctrineParamConverter has such an auto-mapping feature as fallback, until they face weird bugs due to a random object being selected.
And from experience, those bugs can be hard to detect. At Incenteev, I never intended to rely on this auto-mapping. Several places had to force an explicit mapping for the param converter to restrict it to the primary. When migrating to EntityValueResolver with auto-mapping disabled, I discovered some cases where a random entity was loaded (and this was hiding a bug in the code path handling `null` as the CI was never reaching it due to using a random object instead).

**Pull request merged**